### PR TITLE
feat: export html parts renderers

### PR DESCRIPTION
**Pull request by Volodomor**
Pull Request for RFC https://github.com/gravity-ui/rfc/issues/7

### CODE REVIEW

**Comment by obenjiro** [comment]
@korvin89 @ValeraS @sofiushko

I would appreciate it if you could take some time to review this pull request. Thank you.

**Review by ValeraS — CHANGES_REQUESTED** [review]

File: src/utils/generateRenderContent.ts [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+    const content = params.bodyContent ?? {};
+    const {theme, className} = content;
+    const bodyClasses = Array.from(
+        new Set([...getRootClassName(theme), ...(className ? className.split(' ') : [])]),
+    );
+    const bodyContent: BodyContent = {
+        attributes: {
+            class: bodyClasses.filter(Boolean).join(' '),
+        },
+        className: bodyClasses,
```
[the remark this answers is not in the document]
Comment (Volodomor) [reply src/utils/generateRenderContent.ts:63]: Returned to the original version

File: src/types.ts [hunk 2]
```diff
@@ -36,20 +37,27 @@ export interface CommonOptions {
...
-        beforeRoot: string[];
-        root?: string;
-        afterRoot: string[];
-    };
+    inlineScripts: string[];
+    title: string;
+}
+
+export interface BodyContent {
+    className: string[];
```
Comment (ValeraS) [inline src/types.ts:52]:
```suggestion
    /** @deprecated use attributes.class instead */
    className: string[];
    attributes: Attributes;
```

File: src/types.ts [hunk 3]
```diff
@@ -59,13 +67,15 @@ export interface RenderHelpers {
     renderInlineStyle(content: string): string;
     renderMeta(meta: Meta): string;
     renderLink(link: Link): string;
+    attrs(obj: Attributes): string;
 }
 export interface Plugin<Options = any, Name extends string = string> {
     name: Name;
     apply: (params: {
         options: Options | undefined;
         renderContent: RenderContent;
```
Comment (ValeraS) [inline src/types.ts:76]:
```suggestion
        renderContent: Omit<RenderContent, 'bodyContent'> & {bodyContent: OldBodyContent};
```

where 
```ts
type OldBodyContent = Omit<BodyContent, 'attributes'>;
```

File: README.md [hunk 4]
```diff
@@ -438,3 +448,62 @@ app.get((req, res) => {
...
+
+  const content = generateRenderContent(plugins, {
+    title: 'Home page',
+  });
+
+  const {htmlAttributes, helpers, bodyContent} = content;
+
+  const bodyAttributes = {
+    class: bodyContent.className.filter(Boolean).join(' '),
+  };
```
Comment (ValeraS) [inline README.md:484]:
```suggestion
```

File: README.md [hunk 5]
```diff
@@ -438,3 +448,62 @@ app.get((req, res) => {
...
+
+  res.write(`
+        <!DOCTYPE html>
+        <html ${helpers.attrs({...htmlAttributes})}>
+        <head>
+            ${renderHeadContent(content)}
+        </head>
+        <body ${helpers.attrs(bodyAttributes)}>
```
Comment (ValeraS) [inline README.md:492]:
```suggestion
        <body ${helpers.attrs(bodyContent.attributes)}>
```

**Review by ValeraS — CHANGES_REQUESTED** [review]

File: src/utils/generateRenderContent.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (ValeraS) [reply src/utils/generateRenderContent.ts:63]: Let's add attributes to body content after plugins.

**Comment by Volodomor** [comment]
@ValeraS done

**Review by ValeraS — APPROVED** [review]

**Comment by Volodomor** [comment]
@ValeraS выпусти, пожалуйста, новую версию

**Pull request merged**